src/utils/zcl_abapgit_abap_language_vers.clas.abap

```diff
@@ -0,0 +1,144 @@
+        cloud_development    TYPE ty_allowed_abap_langu_version VALUE '5',
+      END OF c_allowed_abap_langu_version .
+
+    METHODS get_abap_language_vers_by_devc
+      IMPORTING
+        !iv_package                     TYPE devclass
+      RETURNING
+        VALUE(rv_abap_language_version) TYPE string .
+    METHODS get_abap_language_vers_by_repo
+      IMPORTING
+        !io_repo                        TYPE REF TO zif_abapgit_repo
+      RETURNING
+        VALUE(rv_abap_language_version) TYPE string .
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_ABAP_LANGUAGE_VERS IMPLEMENTATION.
+
+
+  METHOD get_abap_language_vers_by_devc.
+
+    DATA lv_abap_lang_version_swc TYPE string.
+    DATA lo_abap_language_version_cfg TYPE REF TO object.
+
+    TRY.
+        CREATE OBJECT lo_abap_language_version_cfg TYPE ('IF_ABAP_LANGUAGE_VERSION_CFG').
+
+        CALL METHOD ('CL_ABAP_LANGUAGE_VERSION_CFG')=>('GET_INSTANCE')
+          RECEIVING
+            ro_instance = lo_abap_language_version_cfg.
+
+        CALL METHOD lo_abap_language_version_cfg->('IF_ABAP_LANGUAGE_VERSION_CFG~GET_PACKAGE_DEFAULT_VERSION')
+          EXPORTING
+            iv_package_name             = iv_package
+          RECEIVING
+            rv_default_language_version = lv_abap_lang_version_swc.
+
+        CASE lv_abap_lang_version_swc.
+          WHEN  c_allowed_abap_langu_version-standard.
+            rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-standard.
+          WHEN c_allowed_abap_langu_version-standard_source_code.
+            rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-standard.
+          WHEN c_allowed_abap_langu_version-key_user.
+            rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-key_user.
+          WHEN c_allowed_abap_langu_version-cloud_development.
+            rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-cloud_development.
+          WHEN OTHERS.
+            rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-undefined.
+        ENDCASE.
+      CATCH cx_root.
+        rv_abap_language_version = zif_abapgit_dot_abapgit=>c_abap_language_version-undefined.
+    ENDTRY.
+  ENDMETHOD.
+
+
+  METHOD get_abap_language_vers_by_objt.
+
+    DATA lo_abap_language_version TYPE REF TO object.
+
+    TRY.
+
+        CREATE OBJECT lo_abap_language_version TYPE ('IF_ABAP_LANGUAGE_VERSION').
+
+        CALL METHOD ('CL_ABAP_LANGUAGE_VERSION')=>('GET_INSTANCE')
+          RECEIVING
+            ro_instance = lo_abap_language_version.
+
+        CALL METHOD lo_abap_language_version->('IF_ABAP_LANGUAGE_VERSION~GET_DEFAULT_VERSION')
+          EXPORTING
+            iv_object_type     = iv_object_type
+            iv_package_name    = iv_package
+          RECEIVING
+            rv_default_version = rv_allowed_abap_langu_version.
+
+      CATCH cx_root.
+        rv_allowed_abap_langu_version =  c_allowed_abap_langu_version-standard.
```

Comment:
I guess, here we have to differentiate between source code object and non-source code objects, don't we?

Reply:
@ThomasPloski and I discussed this. We try put the logic to the object handlers that need a "special" standard ABAP language version. If this won't work out, we add this logic later